src/main/java/seedu/address/logic/commands/FilterPetCommand.java

```diff
@@ -0,0 +1,74 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.function.Predicate;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.model.Model;
+import seedu.address.model.pet.Pet;
+
+/**
+ * Filters and lists all pets in address book whose attributes match the argument keywords.
+ * Keyword matching is case-insensitive.
+ */
+public class FilterPetCommand extends Command {
+    public static final String COMMAND_WORD = "filter-p";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Filters all pets who are tagged: "
```

Comment:
We are not using tags are we? So maybe change this line to "Filters all pets with attributes:"